projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx

```diff
@@ -35,10 +35,10 @@ export default createHigherOrderComponent(
 			select => select( 'core/block-editor' ).hasSelectedInnerBlock( clientId, true ),
 			[]
 		);
-
+		const { hasParentBanner } = useContext( PaidBlockContext ) || {};
```

Comment:
Is it okay to call the hook conditionally?

Reply:
It's not :) but you're already calling hook in this component conditionally anyway. Should be fine for now since blocks can't change type.